src/kdbindings/signal.h

```diff
@@ -11,6 +11,11 @@
 
 #pragma once
 
+// Enables compiler warnings for unused ConnectionHandle results.
+#ifndef KDBINDINGS_WARN_UNUSED
+#define KDBINDINGS_WARN_UNUSED [[nodiscard]]
```

Comment:
May as well use [[nodiscard]] directly if just defining it.

Or if you want to make it optional we should make it an option in CMake.

To me it seems strange asking downstream to define KDBINDINGS_WARN_UNUSED if they *don't* want warnings. :)

Reply:
We have to document this somewhere as well, right?